packages/react-next/src/components/FocusTrapZone/FocusTrapZone.tsx

```diff
@@ -139,204 +104,278 @@ export class FocusTrapZone extends React.Component<IFocusTrapZoneProps, {}> impl
         ? firstFocusableSelector
         : firstFocusableSelector && firstFocusableSelector();
 
-    let _firstFocusableChild: HTMLElement | null = null;
+    let firstFocusableChild: HTMLElement | null = null;
 
-    if (this._root.current) {
+    if (root.current) {
       if (focusSelector) {
-        _firstFocusableChild = this._root.current.querySelector('.' + focusSelector);
+        firstFocusableChild = root.current.querySelector('.' + focusSelector);
       }
 
       // Fall back to first element if query selector did not match any elements.
-      if (!_firstFocusableChild) {
-        _firstFocusableChild = getNextElement(
-          this._root.current,
-          this._root.current.firstChild as HTMLElement,
+      if (!firstFocusableChild) {
+        firstFocusableChild = getNextElement(
+          root.current,
+          root.current.firstChild as HTMLElement,
           false,
           false,
           false,
           true,
         );
       }
     }
-    if (_firstFocusableChild) {
-      this._focusAsync(_firstFocusableChild);
-    }
-  }
-
-  private _focusAsync(element: HTMLElement): void {
-    if (!this._isBumper(element)) {
-      focusAsync(element);
+    if (firstFocusableChild) {
+      focusAsync(firstFocusableChild);
     }
-  }
+    // eslint-disable-next-line react-hooks/exhaustive-deps -- internal state will not change
```

Comment:
Probably better to leave the rule enabled and add `internalState` to the deps list--that way the rule will be able to help catch any deps which might be added in the future.